### PR TITLE
Support .env files

### LINKED ISSUE #107: Support `.env` files
This would be a little bit easier than having to set the variables in the shell.

### PR DESCRIPTION
**Pull request by sfc-gh-cnivera**
Resolves #107 

Instead of having to export/set the variables in the user's shell, it'd be handy to support a `.env` file where the prerequisite variables could be set. We still support setting the variables directly in the shell / Python code.

This PR adds the `python-dotenv` package to support .env files, and clarifies the instructions in the README.

## Testing
Set the environment variables in a `.env` file and verify that they are reflected properly in the app.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,34 +79,34 @@ is set, the default is `snowflake`, which uses standard username/password suppor
 
 ```bash
 # no SNOWFLAKE_AUTHENTICATOR needed
-export SNOWFLAKE_PASSWORD="<your-snowflake-password>"
+SNOWFLAKE_PASSWORD="<your-snowflake-password>"
```
Comment (sfc-gh-nsehrawat) [inline README.md:82]: Why did we remove export?  These env vars won't be visible to the model generator without export.

File: semantic_model_generator/snowflake_utils/env_vars.py [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import os
 
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (sfc-gh-nsehrawat) [inline semantic_model_generator/snowflake_utils/env_vars.py:5]: Is this a no-op if the `.env` is absent?

File: README.md (hunk 1 above)
Comment (sfc-gh-cnivera) [reply README.md:82]: Ah, I wanted to match the example syntax to the generic `.env` format and not have OS specific keywords like `export` or `set`. It's a bit unclear though, I'll add example specific for each OS!

File: semantic_model_generator/snowflake_utils/env_vars.py (hunk 2 above)
Comment (sfc-gh-cnivera) [reply semantic_model_generator/snowflake_utils/env_vars.py:5]: Yep!

**Review by sfc-gh-yayin — APPROVED** [review]

**Pull request merged**